File: FSUIKit/Classes/FSShakeBaseController.m
//
//  FSShakeBaseController.m
//  myhome
//
//  Created by FudonFuchina on 2017/12/8.
//  Copyright © 2017年 fuhope. All rights reserved.
//

#import "FSShakeBaseController.h"

@interface FSShakeBaseController ()

@end

@implementation FSShakeBaseController

- (void)viewDidLoad {
    [super viewDidLoad];
    [UIApplication sharedApplication].applicationSupportsShakeToEdit = YES;
    [self becomeFirstResponder];
}

- (void)motionBegan:(UIEventSubtype)motion withEvent:(UIEvent *)event {
}

- (void)motionCancelled:(UIEventSubtype)motion withEvent:(UIEvent *)event {
}

- (void)motionEnded:(UIEventSubtype)motion withEvent:(UIEvent *)event {
    if (event.subtype == UIEventSubtypeMotionShake) {
        [self shakeEndActionFromShakeBase];
    }
    return;
}

- (void)shakeEndActionFromShakeBase{
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: FSUIKit/Classes/Swift/FSBaseController/FSShakeBaseController.swift
//
//  FSShakeBaseController.swift
//  FSBaseController
//
//  Translated from Objective-C
//

import UIKit

@objc open class FSShakeBaseController: FSBaseController {
    
    open override func viewDidLoad() {
        super.viewDidLoad()
        UIApplication.shared.applicationSupportsShakeToEdit = true
        if canBecomeFirstResponder {
            becomeFirstResponder()
        }
    }
    
    open override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        if canResignFirstResponder {
            resignFirstResponder()
        }
    }
    
    open override var canBecomeFirstResponder: Bool {
        return true
    }
    
    open override func motionBegan(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
    }
    
    open override func motionCancelled(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
    }
    
    open override func motionEnded(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
        if event?.subtype == .motionShake {
            shakeEndActionFromShakeBase()
        }
    }
    
    @objc open func shakeEndActionFromShakeBase() {
        // 子类重写此方法处理摇一摇结束事件
    }
}
